stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepositoryImplTest.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:OptIn(ExperimentalCoroutinesApi::class)
+
+package io.getstream.chat.android.offline.repository.domain.channel.internal
+
+import io.getstream.chat.android.offline.randomChannel
+import io.getstream.chat.android.offline.randomMessage
+import io.getstream.chat.android.offline.randomUser
+import io.getstream.chat.android.test.TestCoroutineRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.argThat
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import java.util.Date
+
+internal class ChannelRepositoryImplTest {
+    @get:Rule
+    val testCoroutines: TestCoroutineRule = TestCoroutineRule()
+
+    private val channelDao: ChannelDao = mock()
+    private val channelRepository: ChannelRepositoryImpl =
+        ChannelRepositoryImpl(
+            channelDao,
+            { randomUser() },
+            { randomMessage() }
+        )
+
+    @Test
+    fun `Given channel without messages in DB Should insert channel with updated last message`() = runTest {
+        val channel = randomChannel(messages = emptyList())
+        val lastMessage = randomMessage(createdAt = Date())
+        whenever(channelDao.select(listOf("cid"))) doReturn listOf(channel.toEntity())
+
+        channelRepository.updateLastMessageForChannel("cid", lastMessage)
+
+        verify(channelDao).insert(
+            argThat { channelEntity ->
+                channelEntity.lastMessageAt == lastMessage.createdAt &&
+                    channelEntity.lastMessageId == lastMessage.id
+            }
+        )
+    }
+
+    @Test
+    fun `Given channel with outdated lastMessage in DB Should insert channel with updated last message`() = runTest {
```

Comment:
Should we also add a test where we check that when the date of the "new" message is somehow older than the previous last message, the `channelDao` has no interactions?

Reply:
Makes sense